### DISCUSSION TITLE
Network Client

### DISCUSSION THREAD

**Discussion by rpatel8del** [pointfreeco/swift-composable-architecture#1840] [Q&A]
Has anyone implemented a generic API client that can be injected in as a dependency value?

**Comment by Nicolaidam** [comment]
Yes, you can use .dependency(\.apiClient, apiCLient) on the initial reducer to inject the version of the apiClient you would like. 

Then you can just conform ApiClient to TestDependencyKey since you dont need a live value.

**Reply by Sajjon** [reply]
@Nicolaidam I think OP asked for an existing RESTClient Op can use.
